.github/workflows/test-changed-firestore.yml

```diff
@@ -29,16 +33,104 @@ jobs:
           yarn
       - name: build
         run: yarn build:changed firestore
-      - name: Run tests if firestore or its dependencies has changed
-        run: yarn test:changed firestore
+      - name: Archive build
+        if: ${{ !cancelled() }}
+        run: |
+          tar -cf build.tar .
+          gzip build.tar
+      - name: Upload build archive
+        if: ${{ !cancelled() }}
+        uses: actions/upload-artifact@v3
+        with:
+          name: build.tar.gz
+          path: build.tar.gz
+          retention-days: ${{ env.artifactRetentionDays }}
+
+  compat-test-chrome:
+    name: Test Firestore Compatible
+
+    runs-on: ubuntu-latest
+
+    needs: build
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@master
+        with:
+          # This makes Actions fetch all Git history so run-changed script can diff properly.
+          fetch-depth: 0
+      - name: Set up Node (14)
+        uses: actions/setup-node@v3
+        with:
+          node-version: 14.x
+      - name: install Chrome stable
+        run: |
+          sudo apt-get update
+          sudo apt-get install google-chrome-stable
+      - name: Download build archive
+        uses: actions/download-artifact@v3
+        with:
+          name: build.tar.gz
+      - name: Unzip build artifact
+        run: tar xf build.tar.gz
+      - name: Bump Node memory limit
+        run: echo "NODE_OPTIONS=--max_old_space_size=4096" >> $GITHUB_ENV
+      - name: Test setup and yarn install
+        run: |
+          cp config/ci.config.json config/project.json
+          yarn
+      - name: Run compat tests
+        run: cd packages/firestore-compat && yarn run test:ci
+
+  test-chrome:
+    name: Test Firestore
+    strategy:
+      matrix:
+        test-name: ["test:browser", "test:travis", "test:lite:browser", "test:browser:prod:nameddb", "test:lite:browser:nameddb"]
+
+    runs-on: ubuntu-latest
+
+    needs: build
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@master
+        with:
+          # This makes Actions fetch all Git history so run-changed script can diff properly.
+          fetch-depth: 0
+      - name: Set up Node (14)
+        uses: actions/setup-node@v3
+        with:
+          node-version: 14.x
+      - name: install Chrome stable
+        run: |
+          sudo apt-get update
+          sudo apt-get install google-chrome-stable
+      - name: Download build archive
+        uses: actions/download-artifact@v3
+        with:
+          name: build.tar.gz
+      - name: Unzip build artifact
+        run: tar xf build.tar.gz
+      - name: Bump Node memory limit
+        run: echo "NODE_OPTIONS=--max_old_space_size=4096" >> $GITHUB_ENV
+      - name: Test setup and yarn install
+        run: |
+          cp config/ci.config.json config/project.json
+          yarn
+      - name: Run tests
+        run: cd packages/firestore && yarn run ${{ matrix.test-name }} && cd -
```

Comment:
What is the purpose of `cd -`? Can it be removed?

Reply:
Removed.